core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -952,11 +936,12 @@ private[spark] object MapOutputTracker extends Logging {
   }
 
   // Opposite of serializeMapStatuses.
-  def deserializeMapStatuses(bytes: Array[Byte]): Array[MapStatus] = {
+  def deserializeMapStatuses(bytes: Array[Byte], conf: SparkConf): Array[MapStatus] = {
     assert (bytes.length > 0)
 
     def deserializeObject(arr: Array[Byte], off: Int, len: Int): AnyRef = {
-      val objIn = new ObjectInputStream(new ZstdInputStream(
+      val codec = CompressionCodec.createCodec(conf, "zstd")
```

Comment:
So, this is the main change?

Reply:
Yes. The codec is wrapped in buffered stream which avoids overhead excessive of JNI call while trying to compress small amount of data.